### ISSUE TITLE
[Bug]: setup function imported from '@storybook/vue3' is not working

### ISSUE THREAD

**Issue by methodev** [storybookjs/storybook#23844]
### Describe the bug

Given that I have an import statement in the **preview.js** file as follows:

```
import { setup } from '@storybook/vue3';
```

When I call the setup function as follows:
```
setup((app) => {
  console.log('Here comes the app:', app);
});
```

Then nothing happens. Seems like the function given to the setup as an argument is not being executed.

### System

```shell
Environment Info:

  System:
    OS: macOS 13.4.1
    CPU: (12) x64 Apple M2 Pro
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 115.0.5790.170
    Edge: 114.0.1823.67
    Firefox: 116.0.2
    Safari: 16.5.2
  npmPackages:
    @storybook/addon-a11y: ^7.3.0 => 7.3.0 
    @storybook/addon-actions: ^7.3.0 => 7.3.0 
    @storybook/addon-console: ^2.0.0 => 2.0.0 
    @storybook/addon-essentials: ^7.3.0 => 7.3.0 
    @storybook/addon-interactions: ^7.3.0 => 7.3.0 
    @storybook/addon-links: ^7.3.0 => 7.3.0 
    @storybook/addon-styling: ^1.3.6 => 1.3.6 
    @storybook/addons: ^7.3.0 => 7.3.0 
    @storybook/blocks: ^7.3.0 => 7.3.0 
    @storybook/mdx1-csf: ^1.0.0 => 1.0.0 
    @storybook/testing-library: ^0.2.0 => 0.2.0 
    @storybook/theming: ^7.3.0 => 7.3.0 
    @storybook/vue3: ^7.3.0 => 7.3.0 
    @storybook/vue3-webpack5: ^7.3.0 => 7.3.0
```

### Additional context

I noticed this today in v7.2.3 and shortly after I saw that v7.3.0 is released. So I updated to it to see if this is fixed, but it's still not working.

**Comment by chakAs3** [comment]
@methodev , i think i encountered this recently here a PR with the fix [#23772](https://github.com/storybookjs/storybook/pull/23772)

**Comment by methodev** [comment]
> 

Hey @chakAs3,

Thanks for your response.

Are you sure it could be just this – a duplication of **@storybook/vue3**?

I'm not sure how to verify if there are multiple versions of it. There is only one **vue3** dir inside the **/node_modules/@storybook** dir.

And my devDependencies are as follows:
```
    "@storybook/addon-a11y": "^7.3.2",
    "@storybook/addon-console": "^2.0.0",
    "@storybook/addon-essentials": "^7.3.2",
    "@storybook/addon-interactions": "^7.3.2",
    "@storybook/addon-links": "^7.3.2",
    "@storybook/blocks": "^7.3.2",
    "@storybook/testing-library": "^0.2.0",
    "@storybook/theming": "^7.3.2",
    "@storybook/vue3": "^7.3.2",
    "@storybook/vue3-webpack5": "^7.3.2",
```

And in my **package-lock.json** I find **@storybook/vue3** only four times and all of them are set to **7.3.2**.

And yet:
```
console.log('before SETUP'); // this one is printed
setup((app) => {
  console.log('in SETUP'); // this one isn't
});
```

**Comment by methodev** [comment]
I updated my Storybook to v7.4.0-alpha.1 and it's fine now.
10x anyway. 🙂

**Comment by chakAs3** [comment]
> I updated my Storybook to v7.4.0-alpha.1 and it's fine now. 10x anyway. 🙂
the PR fixed that was merged in v7.4.0-alpha.1
